packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,560 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:developer' as developer;
+import 'dart:math' as math;
+import 'dart:ui' as ui show window, Picture, SceneBuilder, PictureRecorder;
+import 'dart:ui' show Offset;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'binding.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+
+typedef Widget InspectorSelectButtonBuilder(BuildContext context, VoidCallback onPressed);
+
+/// A widget that enables inspecting the child widget's structure.
+///
+/// Select a location on your device or emulator and view what widgets and
+/// render object that best matches the location. An outline of the selected
+/// widget and terse summary information is shown on device with detailed
+/// information is shown in the observatory or in IntelliJ when using the
+/// Flutter Plugin.
+///
+/// The inspector has a select mode and a view mode.
+///
+/// In the select mode, tapping the device selects the widget that best matches
+/// the location of the touch and switches to view mode. Dragging a finger on
+/// the device selects the widget under the drag location but does not switch
+/// modes. Touching the very edge of the bounding box of a widget triggers
+/// selecting the widget even if another widget that also overlaps that
+/// location would otherwise have priority.
+///
+/// In the view mode, the previous;y selected widget is outlined however
+/// touching the device has the same effect it would have if the inspector
+/// wasn't present. This allows interacting with the application and viewing how
+/// the selected widget changes position. Clicking on the select icon in th
+/// bottom left corner of the application switches back to select mode.
+class WidgetInspector extends StatefulWidget {
+  /// Creates a widget that enables inspection for the child.
+  ///
+  /// The [child] argument must not be null.
+  const WidgetInspector({
+    Key key,
+    @required this.child,
+    @required this.selectButtonBuilder,
+  }) : assert(child != null),
+       super(key: key);
+
+  /// The widget that is being inspected.
+  final Widget child;
+
+  /// A builder that is called to create the select button
+  final InspectorSelectButtonBuilder selectButtonBuilder;
+
+  @override
+  WidgetInspectorState createState() => new WidgetInspectorState();
+}
+
+class WidgetInspectorState extends State<WidgetInspector>
```

Comment:
This should probably be private.

Alternatively, it and all its members have to be documented, but that doesn't seem useful to the end user.

You can work around it being private in the test by assigning it to a variable of type "dynamic" and then blindly accessing its members. (Just make sure to leave a comment explaining that it's ok that you're doing such a crazy thing because it's only a test!)